### PR TITLE
Fix various disposable issues

### PR DESCRIPTION
**Pull request by Bykiev**
This PR fixes various disposable issues

### CODE REVIEW

File: src/NATS.Client.Core/Internal/SslClientAuthenticationOptionsExtensions.cs [hunk 1]
```diff
@@ -30,9 +30,9 @@ public static SslClientAuthenticationOptions LoadClientCertFromPem(this SslClien
         // https://github.com/dotnet/runtime/blob/380a4723ea98067c28d54f30e1a652483a6a257a/src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs#L192-L197
         if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
         {
-            var ephemeral = leafCert;
-            leafCert = new X509Certificate2(leafCert.Export(X509ContentType.Pfx));
-            ephemeral.Dispose();
+            var ephemeral = new X509Certificate2(leafCert.Export(X509ContentType.Pfx));
+            leafCert.Dispose();
+            leafCert = ephemeral;
```
Comment (mtmk) [inline src/NATS.Client.Core/Internal/SslClientAuthenticationOptionsExtensions.cs:35]: isn't this essentially the same thing? unless i am missing something.

File: src/NATS.Client.Core/Internal/SslStreamConnection.cs [hunk 2]
```diff
@@ -115,9 +115,14 @@ public void SignalDisconnected(Exception exception)
     public async Task AuthenticateAsClientAsync(NatsUri uri, TimeSpan timeout)
     {
         var options = await _tlsOpts.AuthenticateAsClientOptionsAsync(uri).ConfigureAwait(true);
+        var networkStream = new NetworkStream(_socket, true);
+
 #if NETSTANDARD2_0
+        if (_sslStream != null)
+            _sslStream?.Dispose();
```
Comment (mtmk) [inline src/NATS.Client.Core/Internal/SslStreamConnection.cs:122]: Should we handle any exceptions here?

Comment (Bykiev) [reply src/NATS.Client.Core/Internal/SslStreamConnection.cs:122]: I'm not sure, but there shouldn't be any issues. The only issue is there shouldn't be elvis operator in line 122 as I've already checked for null on line 121

File: src/NATS.Client.Core/Internal/SslClientAuthenticationOptionsExtensions.cs (hunk 1 above)
Comment (Bykiev) [reply src/NATS.Client.Core/Internal/SslClientAuthenticationOptionsExtensions.cs:35]: You're right, I'll rollback this change. Thanks!

**Comment by Bykiev** [comment]
Failing test seems doesn't related to changes in this PR. Can you please rerun it again?

**Comment by mtmk** [comment]
@Bykiev sorry forgot to mention. please sign you commits it's a CNCF requirement. you will have to force push. thank you 💯

**Comment by Bykiev** [comment]
@mtmk, thank you, signed

**Comment by Bykiev** [comment]
It seems I need to sign all the commits, I need a help with it. I've tried this command with no luck:
`git rebase --exec "git commit --amend --no-edit -n -S" -i 5847cea30137afb3e96cc6448136ed77a11db137`

**Comment by Bykiev** [comment]
I've cherry-picked the changes to another branch, https://github.com/nats-io/nats.net/pull/625

**Pull request closed without merging**